### PR TITLE
fix: set the correct destination account name for copy volume if storageAccount is specified in storage class

### PR DESCRIPTION
**Pull request by umagnus**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

fix: set the correct destination account name for copy volume if storageAccount is specified in storage class

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

**Review by andyzhangx — COMMENTED** [review]
/ok-to-test

File: pkg/azurefile/azurefile.go [hunk 1]
```diff
@@ -993,28 +993,44 @@ func (d *Driver) ResizeFileShare(ctx context.Context, subsID, resourceGroup, acc
...
+		dstAccountName = srcAccountName
+	}
 	dstFileShareName := shareOptions.Name
-	if srcFileShareName == "" || dstFileShareName == "" {
-		return fmt.Errorf("srcFileShareName(%s) or dstFileShareName(%s) is empty", srcFileShareName, dstFileShareName)
+	if srcAccountName == "" || srcFileShareName == "" || dstFileShareName == "" {
+		return fmt.Errorf("one or more of srcAccountName(%s), srcFileShareName(%s), dstFileShareName(%s) are empty", srcAccountName, srcFileShareName, dstFileShareName)
+	}
+	var srcAccountSasToken string
+	srcAccountSasToken = dstAccountSasToken
```
Comment (andyzhangx) [inline pkg/azurefile/azurefile.go:1017]: `srcAccountSasToken := dstAccountSasToken`

and then remove L1016

File: pkg/azurefile/azurefile.go [hunk 2]
```diff
@@ -993,28 +993,44 @@ func (d *Driver) ResizeFileShare(ctx context.Context, subsID, resourceGroup, acc
...
+	if srcAccountName != dstAccountName && dstAccountSasToken != "" {
+		srcAccountOptions := &azure.AccountOptions{
+			Name:                srcAccountName,
+			ResourceGroup:       accountOptions.ResourceGroup,
+			SubscriptionID:      accountOptions.SubscriptionID,
+			GetLatestAccountKey: accountOptions.GetLatestAccountKey,
+		}
```
Comment (andyzhangx) [inline pkg/azurefile/azurefile.go:1024]: add a log here

Comment (andyzhangx) [reply pkg/azurefile/azurefile.go:1024]: nvm, not necessary here

File: pkg/azurefile/azurefile.go [hunk 3]
```diff
@@ -993,28 +993,44 @@ func (d *Driver) ResizeFileShare(ctx context.Context, subsID, resourceGroup, acc
...
 	}
 
-	srcPath := fmt.Sprintf("https://%s.file.%s/%s%s", accountName, storageEndpointSuffix, srcFileShareName, accountSASToken)
-	dstPath := fmt.Sprintf("https://%s.file.%s/%s%s", accountName, storageEndpointSuffix, dstFileShareName, accountSASToken)
+	srcPath := fmt.Sprintf("https://%s.file.%s/%s%s", srcAccountName, storageEndpointSuffix, srcFileShareName, srcAccountSasToken)
+	dstPath := fmt.Sprintf("https://%s.file.%s/%s%s", dstAccountName, storageEndpointSuffix, dstFileShareName, dstAccountSasToken)
 
-	return d.copyFileShareByAzcopy(ctx, srcFileShareName, dstFileShareName, srcPath, dstPath, "", accountName, accountName, resourceGroupName, accountSASToken, authAzcopyEnv, secretName, secretNamespace, secrets, accountOptions, storageEndpointSuffix)
+	return d.copyFileShareByAzcopy(ctx, srcFileShareName, dstFileShareName, srcPath, dstPath, "", srcAccountName, dstAccountName, resourceGroupName, srcAccountSasToken, authAzcopyEnv, secretName, secretNamespace, secrets, accountOptions, storageEndpointSuffix)
```
Comment (andyzhangx) [inline pkg/azurefile/azurefile.go:1033]: is `srcAccountSasToken` parameter still needed in this func? it should be already contained in `srcPath`

Comment (umagnus) [reply pkg/azurefile/azurefile.go:1033]: yes, it's for the snapshot format when fall back to use sas token when `Storage File Data Privileged Contributor` role is not assigned.

File: pkg/azurefile/azurefile.go (hunk 1 above)
Comment (umagnus) [reply pkg/azurefile/azurefile.go]: done

File: test/e2e/dynamic_provisioning_test.go [hunk 4]
```diff
@@ -1629,6 +1629,40 @@ var _ = ginkgo.Describe("Dynamic Provisioning", func() {
 		test.Run(ctx, cs, ns)
 	})
 
+	ginkgo.It("should clone a volume from an existing volume and use another storage class [file.csi.azure.com]", func(ctx ginkgo.SpecContext) {
```
Comment (andyzhangx) [inline test/e2e/dynamic_provisioning_test.go:1632]: can you add the similar e2e test to blob csi driver? it should fail now?

**Review by andyzhangx — APPROVED** [review]
/lgtm

**Pull request merged**